lib/oli_web/components/delivery/actions/actions.ex

```diff
@@ -194,6 +213,110 @@ defmodule OliWeb.Components.Delivery.Actions do
     {:noreply, assign(socket, user_role_id: previous_role_id)}
   end
 
+  # Transfer data to another section
+
+  def handle_event("transfer_data_modal", _params, socket) do
+    options_for_select = Sections.get_sections_to_transfer_data(socket.assigns.section)
+
+    modal_assigns = %{
+      title: "Transfer data",
+      id: "transfer_data_modal",
+      ok:
+        JS.push("ok_transfer",
+          target: socket.assigns.myself
+        ),
+      cancel:
+        JS.push("cancel_transfer_modal",
+          target: socket.assigns.myself
+        ),
+      options_for_select: options_for_select
+    }
+
+    modal = fn assigns ->
+      ~F"""
+        <SelectSection.render {...@modal_assigns}/>
+      """
+    end
+
+    send(self(), {:show_modal, modal, modal_assigns})
+
+    {:noreply, assign(socket, modal_assigns: nil)}
+  end
+
+  def handle_event("ok_transfer", %{"section_id" => section_id_to_transfer}, socket) do
+    section_id_to_transfer = String.to_integer(section_id_to_transfer)
+    section_to_transfer = Sections.get_section!(section_id_to_transfer)
+
+    %{
+      section: section,
+      user: user,
+      enrollment: enrollment
+    } = socket.assigns
+
+    result =
+      Repo.transaction(fn ->
+        with {:ok, %Enrollment{} = new_enrollment} <-
```

Comment:
All of the code here inside (and including) the `Repo.transaction` should be moved out of `OliWeb` and into `Oli`, since it represents core application functionality, rather than UI specific functionality.  As a general rule we should be placing as little code as possible in the UX layer, allowing greater code reuse and easier testing.